### PR TITLE
#168 [Content] [Plugin] Create edition plugin

### PR DESCRIPTION
**Pull request by fkroockmann**
Show elements of content with a form associated.
For the moment, i have create edition of content value and contentSet.

Remains to be done:
 - Image
 - Keywords
 - Link

### CODE REVIEW

File: src/tb/apps/content/plugins/edition/main.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+            compareElements: function (contentSet, elements) {
+                var currentElements = contentSet.data.elements,
+                    i,
+                    doSet = false;
+
+                if (currentElements.length !== elements.length) {
+                    doSet = true;
+                } else {
+                    for (i = 0; i < elements.length; i = i + 1) {
```
Comment (mickaelandrieu) [inline src/tb/apps/content/plugins/edition/main.js]: i++ ?

File: src/tb/apps/content/plugins/edition/main.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+                            ContentContainer.clear();
+                            self.popin.hide();
+                        });
+                    });
+                });
+            },
+
+            computeContentSet: function (form) {
+                var self = this,
+                    $form = jQuery('#' + form.id),
```
Comment (ndufreche) [inline src/tb/apps/content/plugins/edition/main.js]: Why did you name the $something ?

File: src/tb/component/formbuilder/form/element/views/form.element.view.content.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        },
+
+        buildPopin: function () {
+            var currentPopin = this.element.popinInstance;
+
+            this.popin = (currentPopin !== undefined) ? PopinManager.createSubPopIn(currentPopin) : PopinManager.createPopIn();
+        },
+
+        bindEvents: function () {
+            jQuery('#bb5-ui').on('click', this.updateBtnId, jQuery.proxy(this.onUpdateClick, this));
```
Comment (ndufreche) [inline src/tb/component/formbuilder/form/element/views/form.element.view.content.js]: Sure of that ?
#bb5-ui is the entire toolbar.

File: src/tb/apps/content/plugins/edition/main.js (hunk 2 above)
Comment (fkroockmann) [inline src/tb/apps/content/plugins/edition/main.js]: The object is a jQuery object and its simply to know it with '$'

File: src/tb/component/formbuilder/form/element/views/form.element.view.content.js (hunk 3 above)
Comment (fkroockmann) [inline src/tb/component/formbuilder/form/element/views/form.element.view.content.js]: Yes, we wiil refactore all file later for fix it

File: src/tb/component/formbuilder/form/views/form.view.js [hunk 4]
```diff
@@ -78,7 +78,7 @@ define(['tb.core.Renderer', 'BackBone', 'jquery'], function (Renderer, Backbone,
                 data = this.computeData(jqueryForm);
 
             if (this.form.isValid()) {
-                this.form.onSubmit(data);
+                this.form.onSubmit(data, this.form);
```
Comment (ndufreche) [inline src/tb/component/formbuilder/form/views/form.view.js]: This don't create bug on other forms ?

Comment (fkroockmann) [inline src/tb/component/formbuilder/form/views/form.view.js]: No, cause other form use only the first parameter

File: src/tb/apps/content/plugins/edition/main.js (hunk 2 above)
Comment (ndufreche) [inline src/tb/apps/content/plugins/edition/main.js]: I'm not very enjoy for this coding style and introduced now.
This is a typing style variable like iCount = o or oSelf = this and we don't do that.

File: src/tb/apps/content/plugins/edition/elements/All.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                    config;
+
+                require('content.repository').find(object.type, object.uid).done(function (content) {
+
+                    config = {
+                        'type': 'content',
+                        'label': object.name,
+                        'object_name': object.name,
+                        'image': content.image
+                    };
```
Comment (ndufreche) [inline src/tb/apps/content/plugins/edition/elements/All.js]: little tips object keys don't need quotes.
```js
config = {
       type: 'content',
       label: object.name,
       object_name: object.name,
       image: content.image
};
```

File: src/tb/component/formbuilder/form/views/form.view.js (hunk 4 above)
Comment (ndufreche) [inline src/tb/component/formbuilder/form/views/form.view.js]: ok :)

File: src/tb/apps/content/plugins/edition/elements/All.js (hunk 5 above)
Comment (fkroockmann) [inline src/tb/apps/content/plugins/edition/elements/All.js]: Yes i know, but i prefer with :)

File: src/tb/apps/content/plugins/edition/main.js (hunk 1 above)
Comment (ndufreche) [inline src/tb/apps/content/plugins/edition/main.js]: This is not a JavaScript problem, Is the think for all languages :
A little gift for you @mickaelandrieu 
++ −−
The increment and decrement operators make it possible to write in an extremely terse style. In languages such as C, they made it possible to write one-liners that could do string copies:
 for (p = src, q = dest; !*p; p++, q++) *q = *p;
They also encourage a programming style that, as it turns out, is reckless. Most of the buffer overrun bugs that created terrible security vulnerabilities were due to code like this.
In my own practice, I observed that when I used ++ and --, my code tended to be too tight, too tricky, too cryptic. So, as a matter of discipline, I don’t use them any more. I think that as a result, my coding style has become cleaner.

File: src/tb/component/formbuilder/form/element/views/form.element.view.content.js (hunk 3 above)
Comment (ndufreche) [inline src/tb/component/formbuilder/form/element/views/form.element.view.content.js]: Ok for that if the other application still working

**Pull request merged**